From ef109d1ee33e1b927e20cf07f8c42e7d71912a0d Mon Sep 17 00:00:00 2001
From: =?utf-8?q?Aur=C3=A9lien=20G=C3=A2teau?= <agateau@kde.org>
Date: Tue, 23 Jun 2009 14:32:00 +0200
Subject: [PATCH] Support for icon_data and image_path hints

image_path still need to look up image in themes
---
--- a/plasma/dataengines/notifications/notificationsengine.cpp
+++ b/plasma/dataengines/notifications/notificationsengine.cpp
@@ -170,6 +170,14 @@ uint NotificationsEngine::Notify(const QString &app_name, uint replaces_id,
     if (hints.contains("image_data")) {
         QDBusArgument arg = hints["image_data"].value<QDBusArgument>();
         image = decodeNotificationSpecImageHint(arg);
+    } else if (hints.contains("image_path")) {
+        image.load(hints["image_path"].toString());
+    } else if (hints.contains("icon_data")) {
+        // This hint was in use in version 0.9 of the spec but has been
+        // replaced by "image_data" in version 0.10. We need to support it for
+        // users of the 0.9 version of the spec.
+        QDBusArgument arg = hints["icon_data"].value<QDBusArgument>();
+        image = decodeNotificationSpecImageHint(arg);
     }
     notificationData.insert("image", image);
 
